db/log_writer.cc

```diff
@@ -146,11 +153,6 @@ Status Writer::EmitPhysicalRecord(RecordType t, const char* ptr, size_t n) {
   Status s = dest_->Append(Slice(buf, header_size));
   if (s.ok()) {
     s = dest_->Append(Slice(ptr, n));
-    if (s.ok()) {
-      if (!manual_flush_) {
-        s = dest_->Flush();
-      }
-    }
   }
   block_offset_ += header_size + n;
```

Comment:
Should we increase the block_offset_ only after the append succeeds?

Reply:
I think it doesn't matter, as any failure will make RocksDB stop writing to the file again?

Reply:
Since we already report an error via status, it is not necessary.